### ISSUE TITLE
Adding MediaItems from MediaController immediately shows unresponsive media notification

### ISSUE THREAD

**Issue by NielsMasdorp** [androidx/media#31]
On app startup, I am adding a couple of `MediaItems` to a `MediaController` which is connected to the `MediaSessionService`. This immediately shows an media notification which is unresponsive (play/pause does nothing except updating the notification buttons but no playback). After starting playback from the `PlayerControlView` in the app, the notification starts to work.

My question: is this expected behaviour or is there something wrong with my configuration? I would expect the media notification to show up as soon as playback starts for the first time, not when the app starts. And if the media notification is supposed to show immediately after adding items and connecting the controls I would expect the media notification to be able the successfully start the playback.

Relevant code:

The `MediaSessionService` implementation:

```kotlin
class StreamService : MediaSessionService() {

    private val getAllStreams: GetAllStreams by inject()

    private val serviceJob = SupervisorJob()
    private val serviceScope = CoroutineScope(Dispatchers.Main + serviceJob)

    private lateinit var urlsMap: Map<String, String>
    private lateinit var player: ExoPlayer
    private lateinit var mediaSession: MediaSession

    override fun onCreate() {
        super.onCreate()
        serviceScope.launch {
            val streams = getAllStreams()
            urlsMap = streams.map { it.id to it.url }.toMap() // cache urls
            initialize()
        }
    }

    override fun onDestroy() {
        player.release()
        mediaSession.release()
        super.onDestroy()
    }

    /**
     * Connect [MediaSession] to this [MediaSessionService]
     */
    override fun onGetSession(controllerInfo: MediaSession.ControllerInfo) = mediaSession

    private fun initialize() {
        player = ExoPlayer.Builder(applicationContext)
            .setAudioAttributes(
                AudioAttributes.Builder()
                    .setUsage(C.USAGE_MEDIA)
                    .setContentType(C.CONTENT_TYPE_MUSIC)
                    .build(), true // Automatic requesting and dropping audio focus
            )
            .setHandleAudioBecomingNoisy(true) // Handle headphones disconnect
            .setWakeMode(C.WAKE_MODE_NETWORK) // Wake+WiFi lock while playing
            .build()

        val intent = Intent(this, MainActivity::class.java)
        val immutableFlag = if (Build.VERSION.SDK_INT >= 23) FLAG_IMMUTABLE else 0
        val pendingIntent = PendingIntent.getActivity(
            this,
            0,
            intent,
            immutableFlag or FLAG_UPDATE_CURRENT
        )

        val builder = MediaSession.Builder(applicationContext, player)
            .setSessionActivity(pendingIntent)

        // Using reflection since relevant fix has not been released
        // https://github.com/androidx/media/commit/2a62a5ee302d694eb9c7099024d13607f6143830
        builder.callPrivateFunc("setMediaItemFiller", CustomMediaItemFiller(streamUrls = urlsMap))
        mediaSession = builder.build()
    }

    class CustomMediaItemFiller(private val streamUrls: Map<String, String>) :
        MediaSession.MediaItemFiller {
        override fun fillInLocalConfiguration(
            session: MediaSession,
            controller: MediaSession.ControllerInfo,
            mediaItem: MediaItem
        ): MediaItem = mediaItem
            .buildUpon()
            .setUri(Uri.parse(streamUrls[mediaItem.mediaId]))
            .build()
    }
}
```

The `MediaController` implementation:

```kotlin
class StreamManagerImpl(
    private val application: Application,
) : StreamManager {

    private lateinit var controllerFuture: ListenableFuture<MediaController>

    private val controller: MediaController?
        get() = if (controllerFuture.isDone) controllerFuture.get() else null

    override fun initialize(streams: List<Stream>, controls: StyledPlayerControls) {
        controllerFuture = MediaController.Builder(
            application,
            SessionToken(application, ComponentName(application, StreamService::class.java))
        ).buildAsync()

        controllerFuture.addListener(
            { initController(streams, controls) },
            MoreExecutors.directExecutor()
        )
    }

    override fun release() = MediaController.releaseFuture(controllerFuture)

    private fun initController(streams: List<Stream>, controls: StyledPlayerControls) {
        controls.player = requireController()
        for (stream in streams) {
            requireController().addMediaItem(
                MediaItem.Builder()
                    .setMediaId(stream.id)
                    .setMediaMetadata(
                        MediaMetadata.Builder()
                            .setArtworkData(toByteArray(stream.smallImgRes), PICTURE_TYPE_OTHER)
                            .setTitle(stream.title)
                            .setArtist(stream.desc)
                            .build()
                    ).build()
            )
        }
    }

    private fun requireController(): MediaController = controller!!
}
```

As you can see, not much is happening except the media items are being added when the app is launched, this triggers the media notification creation.

I'm running this on a Pixel 4a with Android 11. 
Using media3 version: `1.0.0-alpha01`

**Comment by marcbaechinger** [comment]
I think I agree to the expectation that we should not show a notification when in `STATE_IDLE`, but only once the player is prepared and the buttons on the notification work. 

There are some changes in the work regarding handling of notifications by the `MediaSessionService`. We will make sure that this issue will be addressed with these upcoming changes.

Thanks for reporting!

**Comment by marcbaechinger** [comment]
To clarify: To have the notification working you can do 

```
mediaController.prepare()
```

just after adding the media items. This will prepare the player and the notification actions will stat playback. Technically they already work, but they are currently only changing the `playWhenReady` flag that has no effect when in `STATE_IDLE` (not prepared).

**Comment by NielsMasdorp** [comment]
Thanks @marcbaechinger. That makes sense. That will indeed make it useable for now and I hope your previous comment will be implemented soon.

**Comment by marcbaechinger** [comment]
This is fixed at least in the `main` branch and in `1.0.0-beta01` that is released soon. We don't show a notification when in `STATE_IDLE`.

I'm closing this issue. Please re-open if you think this is required.
